linalg/sundials.hpp

```diff
@@ -23,384 +23,433 @@
+
+   /// Compute the matrix-vector product \f$v = M_t x \f$ at time t.
+   static int MassMult2(N_Vector x, N_Vector v, realtype t,
+                        void* mtimes_data);
+
+public:
+   /// Construct a serial wrapper to SUNDIALS' ARKode integrator.
+   /** @param[in] type Specifies the RK method type:
+                       - EXPLICIT - explicit RK method (default)
+                       - IMPLICIT - implicit RK method
+                       - IMEX     - implicit-explicit ARK method */
+   ARKStepSolver(Type type = EXPLICIT);
 
 #ifdef MFEM_USE_MPI
-   /// Construct a parallel ARKODESolver, a wrapper for SUNDIALS' ARKODE solver.
-   /** @param[in] comm  The MPI communicator used to partition the ODE system.
-       @param[in] type  Specifies the #Type of ARKODE solver to construct. */
-   ARKODESolver(MPI_Comm comm, Type type = EXPLICIT);
+   /// Construct a parallel wrapper to SUNDIALS' ARKode integrator.
+   /** @param[in] comm The MPI communicator used to partition the ODE system.
+       @param[in] type Specifies the RK method type:
+                       - EXPLICIT - explicit RK method (default)
+                       - IMPLICIT - implicit RK method
+                       - IMEX     - implicit-explicit ARK method */
+   ARKStepSolver(MPI_Comm comm, Type type = EXPLICIT);
 #endif
 
-   /// Specify the scalar relative and scalar absolute tolerances.
-   void SetSStolerances(double reltol, double abstol);
+   /** @brief Initialize ARKode: calls ARKStepCreate() to create the ARKStep
+       memory and set some defaults.
+
+       If the ARKStep has already been created, it checks if the problem size
+       has changed since the last call to Init(). If the problem is the same
+       then ARKStepReInit() will be called in the next call to Step(). If the
+       problem size has changed, the ARKStep memory is freed and realloced
+       for the new problem size. */
+   /** @param[in] f_ The TimeDependentOperator that defines the ODE system
+
+       @note All other methods must be called after Init(). */
+   void Init(TimeDependentOperator &f_);
+
+   /// Integrate the ODE with ARKode using the specified step mode.
+   /**
+       @param[in,out] x  On output, the solution vector at the requested output
+                         time, tout = @a t + @a dt
+       @param[in,out] t  On output, the output time reached
+       @param[in,out] dt On output, the last time step taken
+
+       @note On input, the values of @a t and @a dt are used to compute desired
+       output time for the integration, tout = @a t + @a dt.
+   */
+   virtual void Step(Vector &x, double &t, double &dt);
+
+   /// Attach an MFEM linear solver solver to ARKode.
+   void UseMFEMLinearSolver();
 
-   /// Set a custom Jacobian system solver for implicit methods.
-   void SetLinearSolver(SundialsODELinearSolver &ls_spec);
+   /// Attach a SUNDIALS linear solver to ARKode.
+   void UseSundialsLinearSolver();
```

Comment:
Similar to the `CVODESolver`, it will be nice to clarify which `MFEM`/`SUNDIALS` solvers will be used when the methods `UseMFEMLinearSolver`/`UseSundialsLinearSolver` are called.